src/linter.js

```diff
@@ -629,4 +633,23 @@ export default class Linter {
       });
   }
 
+  _markBadwordUsage(filename, fileData) {
+    if (fileData) {
+      const words = fileData.split(' ');
+
+      for (let word of words) {
+        // Filter out none a-Z characters to simplify matching with our
+        // badword list.
+        word = word.toLowerCase().replace(/[^a-zA-Z]/, '');
```

Comment:
Quick update: We discussed this in IRC and the regexp needs an update to match *all* non a-z characters. The current one only matches the very first one.